subprojects/diagnostics/src/main/java/org/gradle/api/tasks/diagnostics/AbstractReportTask.java

```diff
@@ -71,6 +72,10 @@ public void generate() {
             @Override
             public void generateReport(Project project) throws IOException {
                 generate(project);
+
+                if (!outputsToConsole()) {
```

Comment:
Yes, I'd probably reverse it and make this check more like "are we going to have a report file?" (I suggested a different name below).